feat(http): Update some mentions of "payload" -> "content" (#35335)
mdn/content: files/en-us/glossary/content_header/index.md

---
title: Content header
slug: Glossary/Content_header
page-type: glossary-definition
---

{{GlossarySidebar}}

> **Content headers** are a group of [HTTP headers](/en-US/docs/Web/HTTP/Headers) that describe the bodies of HTTP messages.

Reviewer: This is relevant because the content does not include the multipart message framing in the payload, if present.

```suggestion
**Content headers** are the group of [HTTP headers](/en-US/docs/Web/HTTP/Headers) that describe the content of the body of an HTTP message, after any message framing in the body has been removed.
```

> They specifically describe the {{Glossary("HTTP Content", "message content")}}, rather than the representation of a resource, meaning the length of message data, which part of the resource is carried in the data (for multi-part messages), any encoding applied for transport, and message integrity checks.

Reviewer: ```suggestion
They specifically describe the {{Glossary("HTTP Content", "message content")}} that is conveyed in the message _as it is transported_, such as the length of the message, the transport encoding, which part of the resource is carried in the data (for multi-part messages), and message integrity check.
This differs from the {{Glossary("Representation header", "Representation headers")}}, which describe the final form of the whole resource after transport and reconstruction.
```

> These headers were defined as "Payload headers" in {{RFC("7231")}}, but "content" is used instead because data contained in HTTP/2 and HTTP/3 frame payloads could be header data, body data, or other control information.

Reviewer: ```suggestion
These headers were defined as "Payload headers" in {{RFC("7231")}}, but are now referred to as "Content headers" because data contained in HTTP/2 and HTTP/3 frame payloads could be header data, body data, or other control information.
```

> Later HTTP specifications don't mention "content headers" or expand on the list of relevant headers, but the semantics remain the same, so this way of classifying headers still applies.

Reviewer: ```suggestion
Later HTTP specifications don't mention "content headers" or expand on the list of relevant headers, but the semantics remain the same, so this way of classifying headers can still be used.
```

> [!NOTE]

> > Content headers are not mutually exclusive with {{Glossary("Representation header", "Representation headers")}} which are typically more relevant to understanding HTTP message semantics.

Reviewer: I don't think you mean this: "typically more relevant to understanding HTTP message semantics." They are more relevant to understanding the resource that you get after the message has been reconstructed.

```suggestion
> There is some overlap between content headers, which describe the content in the current message body, and {{Glossary("Representation header", "representation headers")}}, which describes the format of the final resource that this message is carrying (in part or total).
```

These headers may be present in both HTTP request and response messages and include:

- {{HTTPHeader("Content-Length")}}
- {{HTTPHeader("Content-Range")}}
- {{HTTPHeader("Content-Type")}}
- {{HTTPHeader("Trailer")}}

> - {{HTTPHeader("Content-Encoding")}}

Author: I'm not sure if it really makes sense adding more to this list as the representation headers are more relevant, IMO. 

Per 7231:

```
Header fields that specifically describe the payload, rather than the
   associated representation, are referred to as "payload header
   fields".  Payload header fields are defined in other parts of this
   specification, due to their impact on message parsing.

   +-------------------+----------------------------+
   | Header Field Name | Defined in...              |
   +-------------------+----------------------------+
   | Content-Length    | [Section 3.3.2 of [RFC7230]](https://datatracker.ietf.org/doc/html/rfc7230#section-3.3.2) |
   | Content-Range     | [Section 4.2 of [RFC7233]](https://datatracker.ietf.org/doc/html/rfc7233#section-4.2)   |
   | Trailer           | [Section 4.4 of [RFC7230]](https://datatracker.ietf.org/doc/html/rfc7230#section-4.4)   |
   | Transfer-Encoding | [Section 3.3.1 of [RFC7230]](https://datatracker.ietf.org/doc/html/rfc7230#section-3.3.1) |
   +-------------------+----------------------------+
 ```

Reviewer: If these are content headers they should be in the list. Representation headers should be in the representation list. There is a reasonable expectation from readers that the list is exhaustive.

Author: I think the current list is as accurate as it gets, the other `content-*` headers are not relevant as far as I can see. What do you think?

Reviewer: I think so :-)
